compiler/rustc_ast/src/lib.rs

```diff
@@ -19,6 +19,7 @@
 #![feature(never_type)]
 #![feature(rustdoc_internals)]
 #![feature(stmt_expr_attributes)]
+#![recursion_limit = "256"] // FIXME(nnethercote): remove once #124141 is done
```

Comment:
If this only affects rustdoc, could we gate the increase behind `cfg_attr(doc, ...)` so we don't accidentaly start relying on the higher recursion limit.

Reply:
:+1: